.github/workflows/pytest_unit_pr_approval.yml

```diff
@@ -0,0 +1,39 @@
+name: PyTest - Unit - PR approval
+
+on:
+  pull_request:
+    branches:
+      - staging
+    types: [opened, synchronize, reopened]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.12.2
+
+      - name: Install dependencies
```

Comment:
@sevenearths we should run all the tests with one command. Also, the tests should be located in the root folder of the repo. If we have something like `tests/test_rpc/` then we should do something like `cd tests && pytest .`

Reply:
It might be best to wait for the refactor first